### DISCUSSION TITLE
What is the point of a 2 way Layer?

### DISCUSSION THREAD

**Discussion by tausifcreates** [tokio-rs/axum#2538] [Q&A] [answered]
### Summary

![image](https://github.com/tokio-rs/axum/assets/93582436/d4f5e191-d8d5-4508-8643-66ef97e7c53e)
As you can see from the image. The request passes through 3 layers, and then again, response passes through the same layers. What is the point of this 2 way behavior?

### axum version

0.7.4

**Comment by davidpdrsn** [comment, marked as answer]
So you can modify both the request and response from the same middleware.

**Reply by jplatte** [reply]
... or use information from the request to modify the response

**Reply by tausifcreates** [reply]
Thank you guys!
